src/app/views/workflow-preview/WorkflowPreview.jsx

```diff
@@ -21,11 +22,25 @@ const propTypes = {
 export class WorkflowPreview extends Component {
     constructor(props) {
         super(props);
+        this.state = { cantabularDataset: false };
     }
 
-    handleBackButton = () => {
+    handleBackButton = async () => {
         const previousUrl = url.resolve("../");
-        this.props.dispatch(push(previousUrl));
+        await this.getDatasetType(this.props.params.datasetID);
+        if (this.state.cantabularDataset) {
+            this.props.dispatch(push(previousUrl + "/cantabular"));
+        } else {
+            this.props.dispatch(push(previousUrl));
+        }
```

Comment:
Lines 31 to 35 can be replaced by:

```
this.props.dispatch(push(this.state.cantabularDataset ? `${previousUrl}/cantabular` : previousUrl));
```

But that's a bit gnarly, so you could switch things around and do this:

```
handleBackButton = async () => {
  await this.getDatasetType(this.props.params.datasetID);
  const previousUrl = `${url.resolve("../")}${this.state.cantabularDataset ? /cantabular : ""}`
  this.props.dispatch(push(previousUrl));
}
```

Reply:
Done :white_check_mark: